README_BUILD.md

```diff
@@ -159,3 +162,56 @@ if (0)
 It will try to link against the `call_function` symbol even though it's clearly dead code.
 
 For this reason, when building with MSVC, please build in `RelWithDebInfo` mode. If you try to build in `Debug` mode, it will error.
+
+## Instructions applicable to Microcontroller configurations
+
+### Install Prerequisites/Dependencies
+
+Basic system setup
+```shell
+analog@precision:~$ sudo apt-get update
+analog@precision:~$ sudo apt-get install build-essential
+```
+Install Prerequisites
+```shell
+analog@precision:~$ sudo apt-get install git cmake
+```
+
+Install ARM toolchain for cross-compiling
+```shell
+analog@precision:~$ sudo apt-get install gcc-arm-none-eabi
+```
+
+### Clone
+```shell
+analog@precision:~$ git clone https://github.com/analogdevicesinc/libiio.git
+analog@precision:~$ cd libiio
+```
+
+### Configure & Build
+
+When configuring libiio with cmake, there are a few options that need to be turned on while the rest need to be turned off.
+Cmake Options       | Default | Action                |
+------------------- | ------- | --------------------- |
+`WITH_LIBTINYIIOD`  | OFF | Needs to be turned ON     |
+`WITH_EXTERNAL_BACKEND` | OFF | Needs to be turned ON |
+`NO_THREADS`           | OFF |  Needs to be turned ON |
+
```

Comment:
Is there a reason that you can turn on `WITH_LIBTINYIIOD` but have either `WITH_EXTERNAL_BACKEND` or `NO_THREADS` be off? (I don't think so).

If it is not valid - we shouldn't allow it, and set it by default.

Reply:
The idea here was to show which options users need to turn on. I initially created this table when I had a list of 3 options (external backend, no threads and tinyiiod) but since I reduced it to 1, I guess it can be confusion to have a table of one item.
I can rearrange things.